numpy 2.4 causes some test failures

As of numpy 2.4, the CI tests get errors like this:

`TypeError: only 0-dimensional arrays can be converted to Python scalars`

The numpy 2.4 release notes say this:

**Raise TypeError on attempt to convert array with ndim > 0 to scalar**
Conversion of an array with `ndim > 0` to a scalar was deprecated in NumPy 1.25. Now, attempting to do so raises `TypeError`. Ensure you extract a single element from your array before performing this operation.

There are 4 errors like this:
FAILED tests/test_orbit_phase.py::TestOrbitPhase::test_j0737 - SystemError: <...
FAILED tests/test_templates.py::test_simple_fit_unbinned - ValueError: settin...
FAILED tests/test_utils.py::test_find_dmx - TypeError: only 0-dimensional arr...
FAILED tests/test_utils.py::test_merge_dmx - TypeError: only 0-dimensional ar...

Some may be fixed by updating scipy but others may require a code change in PINT.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -29,7 +29,7 @@ classifiers = [
 ]
 requires-python = ">=3.9"
 dependencies = [
-    "numpy>=1.23.0",
+    "numpy>=1.23.0, <2.4",
     "astropy>=5.0.5",
     "pyerfa",
     "scipy>=1.9.0",

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
-numpy>=1.23.0
+numpy>=1.23.0, <2.4
 astropy>=5.0.5
 pyerfa
 scipy>=1.9.0

--- a/src/pint/models/timing_model.py
+++ b/src/pint/models/timing_model.py
@@ -1055,8 +1055,8 @@ class TimingModel:
         # This is the function to root-find:
 
         def funct(t):
-            nu = self.orbital_phase(t, anom="true")
-            return np.remainder((nu + bbi.omega()).value, 2 * np.pi) - np.pi / 2
+            nu = self.orbital_phase(t, anom="true")[0]
+            return np.remainder((nu + bbi.omega()[0]).value, 2 * np.pi) - np.pi / 2
 
         # Handle the input time(s)
         if isinstance(baryMJD, time.Time):

--- a/src/pint/templates/lcprimitives.py
+++ b/src/pint/templates/lcprimitives.py
@@ -394,7 +394,10 @@ class LCPrimitive:
         return np.asarray([self.p[-1], self.errors[-1]]) if error else self.p[-1]
 
     def set_location(self, loc):
-        self.p[-1] = loc
+        if np.isscalar(loc):
+            self.p[-1] = loc
+        else:
+            self.p[-1] = loc[0]
 
     def get_norm(self, error=False):
         # if error: return np.asarray([self.p[0],self.errors[0]])

--- a/src/pint/utils.py
+++ b/src/pint/utils.py
@@ -1312,7 +1312,7 @@ def find_prefix_bytime(
     indices, r1, r2 = get_prefix_timeranges(model, prefixname)
     matches = np.where((t >= r1) & (t < r2))[0]
     if len(matches) == 1:
-        matches = int(matches)
+        matches = int(matches[0])
     return indices[matches]
 
 

